flow/IThreadPool.h

```diff
@@ -92,7 +92,8 @@ class ThreadReturnPromise : NonCopyable {
 		return promise.getFuture();
 	}
 
-	void send(T const& t) { // Can be called safely from another thread.  Call send or sendError at most once.
+	template <class U>
```

Comment:
Why is this change needed?

Reply:
This will allow us to send Never() to ThreadReturnPromise. Previously we have to send a value that has the same type as the Promise.

Reply:
That makes sense. Thanks!

Related, we should change this API to use perfect forwarding (e.g. https://github.com/apple/foundationdb/blob/6dcc7f8ef0ad396908445af21952ddbf2f377163/flow/flow.h#L897) and add some `std::move`s to `genericactors.actor.h`. Right now, it looks like we are copying every read at least twice just in the thread/promise machinery.